Mathlib/AlgebraicTopology/SimplexCategory/GeneratorsRelations/EpiMono.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.AlgebraicTopology.SimplexCategory.GeneratorsRelations.Basic
+/-! # Epi-mono factorisation in the simplex category presented by generators and relations
+
+This file aims to establish that there is a nice epi-mono factorisation in `SimplexCategoryGenRel`.
+More precisely, we introduce two (inductively-defined) morphism property `P_δ` and `P_σ` that
+single out morphisms that are compositions of `δ i` (resp. `σ i`).
+
+We only define these, and prove various lemmas to help reasoning about them.
+
+## TODOs
+ - Prove that every morphism factors as a `P_σ` followed by a `P_δ`.
+
+-/
+
+namespace SimplexCategoryGenRel
+open CategoryTheory
+
+section EpiMono
+
+/-- `δ i` is a split monomorphism thanks to the simplicial identities. -/
+def splitMonoδ {n : ℕ} (i : Fin (n + 2)) : SplitMono (δ i) where
+  retraction := by
+    induction i using Fin.lastCases with
+    | last => exact σ n
+    | cast i => exact σ i
+  id := by
+    cases i using Fin.lastCases
+    · simp only [Fin.natCast_eq_last, Fin.lastCases_last]
+      exact δ_comp_σ_succ
+    · simp only [Fin.natCast_eq_last, Fin.lastCases_castSucc]
+      exact δ_comp_σ_self
+
+instance {n : ℕ} {i : Fin (n + 2)} : IsSplitMono (δ i) := .mk' <| splitMonoδ i
+
+/-- `δ i` is a split epimorphism thanks to the simplicial identities. -/
+def splitEpiσ {n : ℕ} (i : Fin (n + 1)) : SplitEpi (σ i) where
+  section_ := δ i.castSucc
+  id := δ_comp_σ_self
+
+instance {n : ℕ} {i : Fin (n + 1)} : IsSplitEpi (σ i) := .mk' <| splitEpiσ i
+
+/-- Auxiliary predicate to express that a morphism is purely a composition of `σ i`s. -/
+abbrev P_σ := degeneracies.multiplicativeClosure
+
+/-- Auxiliary predicate to express that a morphism is purely a composition of `δ i`s. -/
+abbrev P_δ := faces.multiplicativeClosure
+
+lemma P_σ.σ {n : ℕ} (i : Fin (n + 1)) : P_σ (σ i) := .of _ (.σ i)
+
+lemma P_δ.δ {n : ℕ} (i : Fin (n + 2)) : P_δ (δ i) := .of _ (.δ i)
+
+/-- All `P_σ` are split epis as composition of such. -/
+lemma isSplitEpi_P_σ {x y : SimplexCategoryGenRel} {e : x ⟶ y} (he : P_σ e) : IsSplitEpi e := by
+  induction he with
+  | of x hx => cases hx; infer_instance
+  | id => infer_instance
+  | comp_of _ _ _ h => cases h; infer_instance
+
+/-- All `P_δ` are split monos as composition of such. -/
+lemma isSplitMono_P_δ {x y : SimplexCategoryGenRel} {m : x ⟶ y} (hm : P_δ m) :
+    IsSplitMono m := by
+  induction hm with
+  | of x hx => cases hx; infer_instance
+  | id => infer_instance
+  | comp_of _ _ _ h => cases h; infer_instance
+
+lemma isSplitEpi_P_σ_toSimplexCategory {x y : SimplexCategoryGenRel} {e : x ⟶ y} (he : P_σ e) :
```

Comment:
This may be a slightly better name:
```suggestion
lemma isSplitEpi_toSimplexCategory_map_of_P_σ {x y : SimplexCategoryGenRel} {e : x ⟶ y} (he : P_σ e) :
```

Reply:
Oh right, I went a bit too fast in f52ef73, apologies.